### PR TITLE
update selectedPoints state when points get deleted

### PR DESCRIPTION
**Pull request by echentw**

### CODE REVIEW

**Comment by echentw** [comment]
This also fixes the bug
"first point in region is selected then deleted, then next point is selected"

File: src/actions/messageActions.ts [hunk 1]
```diff
@@ -105,10 +105,9 @@ export const setMainPoint = (
 };
 
 export interface CombinePointsParams {
-  aboveOrBelow: "above" | "below";
-  point: PointI;
   shape: PointShape;
-  index: number;
+  keepIndex: number;
+  deleteIndex: number;
```
Comment (josephmturner) [inline src/actions/messageActions.ts:110]: Very good - as I was working on normalizing state today, I realized that a change like this had to be done.

File: src/reducers/message.ts [hunk 2]
```diff
@@ -247,45 +247,46 @@ function handleCombinePoints(
   state: MessageState,
   action: Action<CombinePointsParams>
 ): MessageState {
+  const withinBounds = (index: number): boolean => {
+    return index >= 0 && index < state.points[action.params.shape].length;
+  };
+
+  const isQuoted = (index: number): boolean => {
+    return state.points[action.params.shape][index].quotedAuthor !== undefined;
```
Comment (josephmturner) [inline src/reducers/message.ts:255]: What if quotedAuthor is null (or an empty string, etc.) for some reason? Would it be alright to instead return:
`!!state.points[action.params.shape][index].quotedAuthor`

Comment (echentw) [reply src/reducers/message.ts]: Yep, thanks for the catch

**Pull request merged**